parts/linux/cloud-init/artifacts/cse_install.sh

```diff
@@ -268,7 +268,13 @@ downloadContainerdWasmShims() {
     fi
 
     for shim in "${shims_to_download[@]}"; do
-        retrycmd_if_failure 30 5 60 curl -fSLv -o "$containerd_wasm_filepath/containerd-shim-${shim}-${binary_version}-v1" "$containerd_wasm_url/containerd-shim-${shim}-v1" 2>&1 | tee $CURL_OUTPUT | grep -E "^(curl:.*)|([eE]rr.*)$" && (cat $CURL_OUTPUT && exit $ERR_KRUSTLET_DOWNLOAD_TIMEOUT) &
+        output_file="$containerd_wasm_filepath/containerd-shim-${shim}-${binary_version}-v1"
+        download_url="$containerd_wasm_url/containerd-shim-${shim}-v1"
+        retrycmd_if_failure 30 5 60 curl -fSLv -o "$output_file" "$download_url" 2>&1 | tee $CURL_OUTPUT &
```

Comment:
We only want the `curl` command to be running in background (with syntax `&`).
Splitting them into multiple steps for easier debugging and can exit with the error code correctly.